src/document.c

```diff
@@ -1090,6 +1093,17 @@ char_autolink_url(hoedown_buffer *ob, hoedown_document *doc, uint8_t *data, size
 	return link_len;
 }
 
+static size_t
+char_image(hoedown_buffer *ob, hoedown_document *doc, uint8_t *data, size_t offset, size_t size) {
+	size_t ret;
+
+	if (!size || data[1] != '[') return 0;
```

Comment:
Good point. Added a test case and fixed the check. The test passed before the bounds fix since probably in most cases this happens to work anyway.